src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -1,72 +1,112 @@
 package by.custom.utilcalculator.service;
 
+import by.custom.utilcalculator.domain.FileUserProgressStorage;
 import by.custom.utilcalculator.domain.UserProgress;
 import by.custom.utilcalculator.domain.constants.Command;
 import by.custom.utilcalculator.domain.constants.steps.*;
+import by.custom.utilcalculator.exception.UserFileNotFoundException;
 
 public class UserProgressManager {
-    //private final UserProgress userProgress;
     private final MessagesCreator messagesCreator;
+    private final FileUserProgressStorage fileUserProgressStorage;
```

Comment:
Please, before using any class think "Can I use abstraction?"
```suggestion
    private final IUserProgressStorage userProgressStorage;
```

Reply:
OK, thats a great trick. Before creating an object ask myself "is it possible to use abstr.?"
Thank you